### ISSUE TITLE
Exception: ('Invalid merge mode:', 'join')   

### ISSUE THREAD

**Issue by ghost** [maciejkula/triplet_recommendations_keras#4]
Hi,

When running with latest Keras, I have the following message: 

  File "<ipython-input-16-748c8f33187e>", line 1, in <module>
    model = get_graph(num_users, num_items, 256)

  File "<ipython-input-5-e0c43137775f>", line 15, in get_graph
    merge_mode='join')

  File "D:_devs\Python01\WinPython-64-2710\python-2.7.10.amd64\lib\site-packages\keras\legacy\models.py", line 241, in add_shared_node
    raise Exception('Invalid merge mode:', merge_mode)

Exception: ('Invalid merge mode:', 'join')

Any idea ?

Thanks

**Comment by eshijia** [comment]
You should use keras 0.3.2 to run this code. For triplet prediction, I think this is a good example to follow.

**Comment by maciejkula** [comment]
I'll update this to the newest Keras at some point. For the moment, try with 0.3.2.

**Comment by Kyung-Min** [comment]
I forked this project and changed some code.
Now, it works for me.
https://github.com/Kyung-Min/triplet_recommendations_keras

**Comment by eshijia** [comment]
@Kyung-Min Your running result of your code seems not right. In addition, you current code should import the `Activation` first. The result is shown as follows:

AUC before training 0.503034233801
Epoch 0
Train on 49906 samples, validate on 5469 samples
Epoch 1/1
2s - loss: -5.0171e-01 - val_loss: -5.0393e-01
AUC 0.501176010872
Epoch 1
Train on 49906 samples, validate on 5469 samples
Epoch 1/1
2s - loss: -5.0882e-01 - val_loss: -5.0773e-01
AUC 0.500396098802
Epoch 2
Train on 49906 samples, validate on 5469 samples
Epoch 1/1
2s - loss: -5.1569e-01 - val_loss: -5.1147e-01
AUC 0.499127734927
Epoch 3
Train on 49906 samples, validate on 5469 samples
Epoch 1/1
2s - loss: -5.2249e-01 - val_loss: -5.1514e-01
AUC 0.498307014854
Epoch 4
Train on 49906 samples, validate on 5469 samples
Epoch 1/1
2s - loss: -5.2920e-01 - val_loss: -5.1877e-01
AUC 0.497376781555

Does the result in your README using the same code?

**Comment by maciejkula** [comment]
Updated with https://github.com/maciejkula/triplet_recommendations_keras/pull/8
